static/js/common/upload-addon.js

```diff
@@ -429,7 +429,8 @@
                     $("<strong>").text(message).appendTo(upload_results);
 
                     // Specific messages for unlisted addons.
-                    if (isUnlisted()) {
+                    var validation_type = results.validation.detected_type
+                    if (["extension", "dictionary", "languagepack"].includes(validation_type) && isUnlisted()) {
```

Comment:
because `.includes` is quite a new addition to JavaScript can you change to use `.indexOf` instead?  Sorry!